sdk/ai/ai-projects/src/agents/files.ts

```diff
@@ -31,6 +31,23 @@ export async function uploadFile(
   return result.body; 
 }
 
+/** Uploads a file for use by other operations. */
+export async function uploadFileAndPoll(
+context: Client, options: UploadFileParameters, sleepInterval: number,
+): Promise<OpenAIFileOutput> {
+  const result = await context.path("/files").post(options);
+  if (!expectedStatuses.includes(result.status)) {
+      throw createRestError(result);
+  }
+  while (result.body.status === "uploaded" || result.body.status === "pending" || result.body.status === "running") {
+    await new Promise(resolve => setTimeout(resolve, sleepInterval));
```

Comment:
Other test code uses delay from @azure/core-util do this would just be `await delay(sleepInterval * 1000)` (delay uses miliseconds but python code takes in seconds for the sleep)